packages/forms/src/documents/pdf/extract.ts

```diff
@@ -1,23 +1,57 @@
-import * as pdfLib from 'pdf-lib';
+import {
+  PDFDocument,
+  PDFName,
+  PDFDict,
+  PDFTextField,
+  PDFField,
+  PDFCheckBox,
+  PDFDropdown,
+  PDFOptionList,
+  PDFRadioGroup,
+} from 'pdf-lib';
 
 import { stringToBase64 } from '../util';
 import type { DocumentFieldValue, DocumentFieldMap } from '../types';
 
+// TODO: copied from pdf-lib acrofield internals, check if it's already exposed outside of acroform somewhere
+export const getWidgets = async (pdfDoc: PDFDocument): Promise<PDFDict[]> => {
+  return pdfDoc.context
+    .enumerateIndirectObjects()
+    .map(([, obj]) => obj)
+    .filter(
+      obj =>
+        obj instanceof PDFDict &&
+        obj.get(PDFName.of('Type')) === PDFName.of('Annot') &&
+        obj.get(PDFName.of('Subtype')) === PDFName.of('Widget')
+    )
+    .map(obj => obj as PDFDict);
+};
```

Comment:
Seems useful, so I'm exporting, but I don't actually feel strongly about it. It's ripped straight from pdf-lib, but seems to be only exposed on the acrofield object, which isn't useful in our case...